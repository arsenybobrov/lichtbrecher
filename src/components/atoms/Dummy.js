import React, { useContext, useState, useEffect } from 'react';
import styled, { ThemeContext } from 'styled-components';
import { lighten } from 'polished';

const Title = styled.h1`
  color: ${(props) => (lighten(0.5, props.theme.colors.primary))};
  background-color: ${(props) => props.active ? 'red' : 'transparent'};
`;

const Dummy = ({ globalConfig, data }) => {
  const theme = useContext(ThemeContext);
  const [mounted, mount] = useState(false);
  const [clicked, toggleClick] = useState(mounted);

  useEffect(() => {
    setTimeout(() => {
      mount(true);
    }, 3000);
  }, []);

  return (
    <>
      {globalConfig && <Title active={clicked}>Global: {globalConfig.data.display_name[0].text}</Title>}
      {data && <Title active={clicked}>Data: {data.data.display_name[0].text}</Title>}
      {mounted && <button onClick={() => toggleClick(!clicked)}>Click me, my color is: {theme.colors.primary}</button>}
    </>
  );
};

export default Dummy;
