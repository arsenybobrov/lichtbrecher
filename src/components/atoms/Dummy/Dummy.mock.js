const DummyMock = {
  globalConfig: {
    data: {
      display_name: [
        { text: 'Dummy text' },
      ],
    },
  },
  data: {
    data: {
      display_name: [
        { text: 'Dummy text' },
      ],
    },
  },
};

export default DummyMock;
