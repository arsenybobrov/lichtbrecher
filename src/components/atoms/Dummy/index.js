import React, { useContext, useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import styled, { ThemeContext } from 'styled-components';
import { lighten } from 'polished';
import { GlobalConfigContext } from '../../../contexts/globalConfig';
import { BreakpointContext } from '../../../contexts/breakpoint';
import mq from '../../../styles/mediaQueries';

const Title = styled.h1`
  color: ${(props) => (lighten(0.5, props.theme.colors.primary))};
  background-color: ${(props) => (props.active ? 'red' : 'transparent')};

  @media (${mq.mdDown}) {
    background-color: black;
    color: red;
  }
`;

const Dummy = ({ data }) => {
  const theme = useContext(ThemeContext);
  const content = useContext(GlobalConfigContext);
  const breakpoint = useContext(BreakpointContext);
  const [mounted, mount] = useState(false);
  const [clicked, toggleClick] = useState(mounted);

  useEffect(() => {
    setTimeout(() => {
      mount(true);
    }, 3000);
  }, []);

  return (
    <>
      {
        content &&
        <Title active={clicked}>Global: {content.data.display_name[0].text}</Title>
      }
      {
        data &&
        <Title active={clicked}>Data: {data.data.display_name[0].text}</Title>
      }
      <p>BREAKPOINT IS: {breakpoint}</p>
      {
        mounted &&
        <button type="button" onClick={() => toggleClick(!clicked)}>
          Click me, my color is: {theme.colors.primary}
        </button>
      }
    </>
  );
};

Dummy.propTypes = {
  data: PropTypes.object,
};

export default Dummy;
