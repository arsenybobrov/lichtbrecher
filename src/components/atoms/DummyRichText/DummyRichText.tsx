import React from 'react';
import uniqueId from 'lodash/uniqueId';

const RichTextElementMap: Record<string, string> = {
  paragraph: 'p',
};

export interface RichTextElementType {
  type: string;
  text: string;
  key?: string;
}

interface RichTextProps {
  content: Array<RichTextElementType>;
}

const renderRichTextElement = ({ type, text, key }: RichTextElementType) => {
  if (type in RichTextElementMap) {
    return React.createElement(RichTextElementMap[type], { key }, text);
  }
  return <pre key={key}>{JSON.stringify({ type, text })}</pre>;
};

const DummyRichText: React.FC<RichTextProps> = ({ content }) => (
  <>
    {content.map((elem) => (
      renderRichTextElement({ ...elem, key: uniqueId(elem.type) })
    ))}
  </>
);

export default DummyRichText;
