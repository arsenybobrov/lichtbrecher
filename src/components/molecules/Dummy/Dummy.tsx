import React from 'react';
import uniqueId from 'lodash/uniqueId';
import DummyHeadline, { HeadlineProps } from '../../atoms/DummyHeadline/DummyHeadline';
import DummyRichText, { RichTextElementType } from '../../atoms/DummyRichText/DummyRichText';


interface PrismicDummyContent {
 headline: Array<HeadlineProps>;
 richtext: Array<RichTextElementType>;
}

interface DummyProps {
  primary: PrismicDummyContent;
}

const Dummy: React.FC<DummyProps> = ({ primary }) => {
  const { headline, richtext } = primary;
  return (
    <>
      {headline.map((headlineProps) => <DummyHeadline {...headlineProps} key={uniqueId('headline')} />)}
      <DummyRichText content={richtext} />
    </>
  );
};

export default Dummy;
