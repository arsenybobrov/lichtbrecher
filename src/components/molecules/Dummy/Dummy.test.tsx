import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { headingMock } from '../../atoms/DummyHeadline/DummyHeadline.test';
import { paragraphsMock } from '../../atoms/DummyRichText/DummyRichText.test';
import Dummy from './Dummy';

const dummyMock = {
  primary: {
    headline: [headingMock(1)],
    richtext: paragraphsMock,
  },
};

describe('render Dummy molecule snapshot', () => {
  // eslint-disable-next-line jest/prefer-expect-assertions
  it('renders Dummy molecule', () => {
    const tree = renderer.create(<Dummy {...dummyMock} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
