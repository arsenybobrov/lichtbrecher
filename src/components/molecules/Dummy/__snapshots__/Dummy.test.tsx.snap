// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nextLink snapshot RichText with paragraphs 1`] = `
Array [
  <p>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit.
  </p>,
  <p>
    Aliquam dui turpis, ultricies quis semper vel, facilisis in leo.
  </p>,
  <p>
    In aliquet orci et egestas sagittis.
  </p>,
  <p>
    Phasellus pretium, elit ac pellentesque porta, lacus sapien laoreet neque, nec bibendum odio leo sollicitudin tellus.
  </p>,
]
`;

exports[`nextLink snapshot Richtext with unordered list 1`] = `
Array [
  <pre>
    {"type":"list-item","text":"first"}
  </pre>,
  <pre>
    {"type":"list-item","text":"second"}
  </pre>,
  <pre>
    {"type":"list-item","text":"third"}
  </pre>,
]
`;

exports[`nextLink snapshot h1 headline 1`] = `
<h1>
  this is a h1 headline
</h1>
`;

exports[`nextLink snapshot h2 headline 1`] = `
<h2>
  this is a h2 headline
</h2>
`;

exports[`nextLink snapshot h3 headline 1`] = `
<h3>
  this is a h3 headline
</h3>
`;

exports[`nextLink snapshot h4 headline 1`] = `
<h4>
  this is a h4 headline
</h4>
`;

exports[`nextLink snapshot h5 headline 1`] = `
<h5>
  this is a h5 headline
</h5>
`;

exports[`nextLink snapshot h6 headline 1`] = `
<h6>
  this is a h6 headline
</h6>
`;

exports[`render Dummy molecule snapshot renders Dummy molecule 1`] = `
Array [
  <h1>
    this is a h1 headline
  </h1>,
  <p>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit.
  </p>,
  <p>
    Aliquam dui turpis, ultricies quis semper vel, facilisis in leo.
  </p>,
  <p>
    In aliquet orci et egestas sagittis.
  </p>,
  <p>
    Phasellus pretium, elit ac pellentesque porta, lacus sapien laoreet neque, nec bibendum odio leo sollicitudin tellus.
  </p>,
]
`;
